packages/grid/x-data-grid-pro/src/tests/rowPinning.DataGridPro.test.tsx

```diff
@@ -765,4 +765,15 @@ describe('<DataGridPro /> - Row pinning', () => {
 
     expect(screen.getByRole('grid').getAttribute('aria-rowcount')).to.equal(`${rowCount + 1}`); // +1 for header row
   });
+
+  // https://github.com/mui/mui-x/issues/5845
+  it('should work with `getCellClassName` when `rows=[]`', () => {
+    const className = 'test-class-name';
+    render(
+      <BaselineTestCase rowCount={2} colCount={1} rows={[]} getRowClassName={() => className} />,
+    );
+
+    expect(getRowById(0)!.classList.contains(className)).to.equal(true);
+    expect(getRowById(1)!.classList.contains(className)).to.equal(true);
```

Comment:
Much simpler.

```suggestion
    expect(getRowById(0)!).to.have.class(className);
    expect(getRowById(1)!).to.have.class(className);
```

Reply:
I did this intentionally to have meaningful error message when test fails.
See https://github.com/mui/mui-x/pull/3607#discussion_r784785626

Reply:
Update: seems to be fixed by [`cd16a06` (#5851)](https://github.com/mui/mui-x/pull/5851/commits/cd16a065e5d76d708ad563f653c59a7dd267e23e)

Reply:
I've commented the related issue in mocha: https://github.com/mochajs/mocha/issues/4910

Reply:
Once mocha fixes this issue we can use the other syntax here, keeping the same convention like other tests.